File: backend/src/main/java/com/example/backend/service/FollowService.java
package com.example.backend.service;

public interface FollowService {
    // theo dõi người dùng
    Boolean followUser(Long followerId, Long followingId);

    // bỏ theo dõi người dùng
    Boolean unfollowUser(Long followerId, Long followingId);

    // Lấy danh sách người theo dõi mình (Followers)
    org.springframework.data.domain.Page<com.example.backend.model.respon.UserResponse> getFollowers(Long userId, int page, int size);

    // Lấy danh sách người mình đang theo dõi (Following)
    org.springframework.data.domain.Page<com.example.backend.model.respon.UserResponse> getFollowing(Long userId, int page, int size);
}

File: frontend/src/service/impl/FollowService.ts
import { http } from "../http/http";
import type { ApiDataResponse, Page, UserResponse } from "../../utils/ResponType";

export const followUser = (currentUserId: number, followingId: number) => {
    return http.post<ApiDataResponse<boolean>>(`/follows/${followingId}?currentUserId=${currentUserId}`);
};

export const unfollowUser = (currentUserId: number, followingId: number) => {
    return http.delete<ApiDataResponse<boolean>>(`/follows/${followingId}?currentUserId=${currentUserId}`);
};

export const getFollowers = (currentUserId: number, page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<UserResponse>>>(
        `/follows/followers?currentUserId=${currentUserId}&page=${page}&size=${size}`
    );
};

export const getFollowing = (currentUserId: number, page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<UserResponse>>>(
        `/follows/following?currentUserId=${currentUserId}&page=${page}&size=${size}`
    );
};
